scylla-cdc-printer/src/main/java/com/scylladb/cdc/printer/Main.java

```diff
@@ -90,83 +99,129 @@ public static void main(String[] args) {
-                    cellValue, (++columnIndex == nonCdcColumnDefinitions.size()));
+        int row = 0;
+        for (RawChange c : rows) {
+            if (type == Type.Event) {
+                if (row++ > 0) {
+                    printSeparator();
+                }
+                OperationType t = c.getOperationType();
+                if (t != last) {
+                    prettyPrintChange(t);
+                }
+                last = t;
+            }
+
+            int columnIndex = 0; // For pretty printing.
+
+            for (ChangeSchema.ColumnDefinition columnDefinition : nonCdcColumnDefinitions) {
+                String columnName = columnDefinition.getColumnName();
+
+                // We can get information if this column was a part of primary
+                // key
+                // in the base table. Note that in CDC log table different
+                // columns
+                // are part of primary key (cdc$stream_id, cdc$time,
+                // batch_seq_no).
+                ChangeSchema.ColumnType baseTableColumnType = columnDefinition.getBaseTableColumnType();
+
+                // Get the information about the data type (as present in CDC
+                // log).
+                ChangeSchema.DataType logDataType = columnDefinition.getCdcLogDataType();
+
+                // Finally, we can get the value of this column:
+                Cell cell = c.getCell(columnName);
+
+                // Depending on the logDataType, you will want
+                // to use different methods of Cell, for example
+                // cell.getInt() if column is of INT type:
+                //
+                // Integer value = cell.getInt();
+                //
+                // getInt() can return null, if the cell value
+                // was NULL.
+                //
+                // For printing purposes, we use getAsObject():
+                Object cellValue = cell.getAsObject();
+
+                prettyPrintCell(columnName, baseTableColumnType, logDataType, cellValue,
+                        (++columnIndex == nonCdcColumnDefinitions.size()));
+            }
         }
 
         prettyPrintEnd();
     }
 
     // Some pretty printing helpers:
 
-    private static void prettyPrintChangeHeader(StreamId streamId, ChangeTime changeTime,
-                                                RawChange.OperationType operationType) {
+    private static void prettyPrintChangeHeader(Type type, StreamId streamId, ChangeTime changeTime) {
         byte[] buf = new byte[16];
         streamId.getValue().duplicate().get(buf, 0, 16);
 
-        System.out.println("┌────────────────── Scylla CDC log row ──────────────────┐");
+        switch (type) {
+        case Row:
+            System.out.println("┌────────────────── Scylla CDC log row ──────────────────┐");
+            break;
+        case Event:
+            System.out.println("┌───────────────── Scylla CDC log event -────────────────┐");
```

Comment:
Small typo "-" instead of "─" (U+2500). (I am aware that it might be not great to have such "strange" symbols in source code).